_posts/2016-11-25-CSAPP-Chapter2.md

```diff
@@ -97,25 +97,70 @@ while now the most significant bit comes first is called big endian.
 
 2.2 Integer Representations
 ---------------------------
+Assume we have an integer data type of w bits. We write a bit vector, or as 
+[xw−1xw−2..x1][x_{w-1}x_{w-2}..x_1] to denote its individual bits. When we refer to 
+such a unsigned number, the value of the number is 
 
-Two’s-Complement Encodings
-The most com-mon computer representation of signed numbers is known as two’s-complement form. 
-This is defined by interpreting the most significant bit of the word to have negative weight. 
-We express this interpretation as a function B2Tw(for “binary to two’s-complement” length w ):
+B2Ux=∑i=0w−1xi∗2iB2U_x = \sum\_{i =0}^{w-1}{x_i\*2^i} 
 
+Then in this case, the max number of a w-bit number is [11111..11], while the min number
+is [000…00]. However, for many applications, we wish to represent negative values as well.
+The most common computer representation is called two’s complement form. This defined by 
+interpretation the most significant bit of the word to served as the sign bit. 
 
+The function is here below: 
+
+B2Tw(x)=−xw−12w−1+∑i=0w−2xi2iB2T_w(x) =-x\_{w-1}2^{w-1}+\sum\_{i =0}^{w-2}x_i2^i 
```

Comment:
Add this to `_include/header.html`

```html
<script type="text/javascript" async
  src="https://cdn.mathjax.org/mathjax/latest/MathJax.js?config=TeX-MML-AM_CHTML">
</script>
```

Then you will get symbols for clause like this `\sum\_{i =0}^{w-1}{x_i\*2^i}`, this is called [LaTex](https://en.wikipedia.org/wiki/LaTeX)